src/store.ts

```diff
@@ -453,6 +458,83 @@ export default function Store(
 
         mpInstance._Persistence.savePersistence(this.persistenceData);
     };
+
+    this.getFirstSeenTime = (mpid: MPID) => {
+        if (!mpid) {
+            return null;
+        }
+
+        if (
+            this.persistenceData &&
+            this.persistenceData[mpid] &&
+            this.persistenceData[mpid].fst
+        ) {
```

Comment:
```suggestion
        if (this.persistenceData?[mpid]?.fst) {
```

Reply:
or maybe could it be as simple as 
```
return this.persistenceData?[mpid]?.fst || null
```
?

Reply:
I think I tried this in a previous revision and it was giving me inconsistent results. My thought was for this PR, I would copy the existing code as is, and then optimize it later.

Reply:
SGTM